### PR TITLE
DepthToSpace - revise reference implementation

### PR DESCRIPTION
**Pull request by pelszkow**
### Details:
 - move `DepthToSpace::evaluate_depth_to_space` to separate standalone function in `runtime::reference::depth_to_space`
 - drop needless reshapes from reference implementation (reshapes doesn't affect elements in tensor, shape was only changed)
 - move backend test to `ngraph/test/backend/depth_to_space.in.cpp`
 - extend backend test set
 - add serialization test
 - add visitor API test

### Tickets:
 - *44624*

### CODE REVIEW

File: ngraph/core/reference/src/runtime/reference/depth_to_space.cpp [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    {
+        namespace reference
+        {
+            void depth_to_space(const char* in,
+                                const Shape& in_shape,
+                                char* out,
+                                const Shape& out_shape,
+                                const std::size_t block_size,
+                                const op::DepthToSpace::DepthToSpaceMode mode,
+                                const size_t elem_size)
```
Comment (ilyachur) [inline ngraph/core/reference/src/runtime/reference/depth_to_space.cpp:23]: nitpick: combination of `std::size_t` and `size_t`

File: ngraph/core/reference/src/runtime/reference/depth_to_space.cpp [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+                //
+                // First and third step doesn't change input data in memory, it change only the
+                // shape of input. From data layout perspective firs and third step may be
+                // omit. The second operation have to be perform on input data with dispared
+                // shape (x').
+                const size_t n_dim = in_shape.at(0);
+                const size_t c_dim = in_shape.at(1);
+                const size_t spatial_dim_index = 2;
+                const size_t spatial_dims = in_shape.size() - spatial_dim_index;
+                const auto c_dim_divider = static_cast<int>(std::pow(block_size, spatial_dims));
```
Comment (ilyachur) [inline ngraph/core/reference/src/runtime/reference/depth_to_space.cpp:39]: Looks like `c_dim_divider` can have `size_t` type.

File: ngraph/core/reference/src/runtime/reference/depth_to_space.cpp (hunk 1 above)
Comment (pelszkow) [reply ngraph/core/reference/src/runtime/reference/depth_to_space.cpp:23]: stick to `size_t`

File: ngraph/core/reference/src/runtime/reference/depth_to_space.cpp (hunk 2 above)
Comment (pelszkow) [reply ngraph/core/reference/src/runtime/reference/depth_to_space.cpp]: switch to `size_t`

File: ngraph/core/include/ngraph/op/depth_to_space.hpp [hunk 3]
```diff
@@ -65,11 +65,6 @@ namespace ngraph
             protected:
```
Comment (jdanieck) [inline ngraph/core/include/ngraph/op/depth_to_space.hpp:65]: Please check includes in this file. Looks like #include `ngraph/op/util/fused_op.hpp` shouldn't be needed.

**Review by jdanieck — APPROVED** [review]

File: ngraph/core/include/ngraph/op/depth_to_space.hpp (hunk 3 above)
Comment (pelszkow) [reply ngraph/core/include/ngraph/op/depth_to_space.hpp:65]: OK

**Review by jdanieck — APPROVED** [review]

File: ngraph/test/backend/depth_to_space.in.cpp [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+#include "util/test_case.hpp"
+#include "util/test_control.hpp"
+
+using namespace ngraph;
+
+static std::string s_manifest = "${MANIFEST}";
+
+using TestEngine = test::ENGINE_CLASS_NAME(${BACKEND_NAME});
+
+NGRAPH_TEST(${BACKEND_NAME}, depth_to_space_block_first_K1_BS2)
```
Comment (ilyachur) [inline ngraph/test/backend/depth_to_space.in.cpp:18]: Can we add SLT tests for depth_to_space operation?

Comment (pelszkow) [reply ngraph/test/backend/depth_to_space.in.cpp:18]: This one:
- https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/tests/functional/shared_test_classes/include/shared_test_classes/single_layer/depth_to_space.hpp
- https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/depth_to_space.cpp

**Review by ilyachur — APPROVED** [review]

**Pull request merged**